### PR TITLE
K8SPXC-720: add pitr test

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPXC-720](https://jira.percona.com/browse/K8SPXC-720) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

1. Add additional PITR test
2. Allow to set `timeBetweenUploads` as float value, instead of integer

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| scheduled-backup | passed |
| upgrade-haproxy | passed |
| storage | passed |
| scaling | passed |
| recreate | passed |
| haproxy | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| self-healing | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| affinity | passed |
| operator-self-healing | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | failed |
| users | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| demand-backup-encrypted-with-tls | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/682c69ae465e2301c211d5756ff84b1ba4bdea51
image: `perconalab/percona-xtradb-cluster-operator:PR-865-682c69ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| scaling-proxysql | passed |
| affinity | passed |
| security-context | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| storage | passed |
| users | passed |
| self-healing | passed |
| tls-issue-self | passed |
| self-healing-advanced | passed |
| tls-issue-cert-manager | passed |
| operator-self-healing | passed |
| tls-issue-cert-manager-ref | passed |
| scheduled-backup | passed |
| validation-hook | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | failed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/682c69ae465e2301c211d5756ff84b1ba4bdea51
image: `perconalab/percona-xtradb-cluster-operator:PR-865-682c69ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scheduled-backup | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| recreate | passed |
| tls-issue-cert-manager | passed |
| restore-to-encrypted-cluster | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup | passed |
| validation-hook | passed |
| pitr | failed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/682c69ae465e2301c211d5756ff84b1ba4bdea51
image: `perconalab/percona-xtradb-cluster-operator:PR-865-682c69ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| upgrade-haproxy | passed |
| big-data | passed |
| recreate | passed |
| storage | passed |
| scheduled-backup | passed |
| scaling | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| init-deploy | passed |
| self-healing-advanced | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | passed |
| pitr | failed |
| one-pod | passed |
| auto-tuning | passed |
| upgrade-consistency | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/23d0e757faada8f947cb987568fe4417dd27fd7e
image: `perconalab/percona-xtradb-cluster-operator:PR-865-23d0e757`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| scheduled-backup | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| haproxy | passed |
| init-deploy | passed |
| upgrade-haproxy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/23d0e757faada8f947cb987568fe4417dd27fd7e
image: `perconalab/percona-xtradb-cluster-operator:PR-865-23d0e757`

**Review by tplavcic — APPROVED** [review]

**Comment by tplavcic** [comment]
I see you gave up on sysbench 😄

File: e2e-tests/pitr/run [hunk 1]
```diff
@@ -108,6 +117,15 @@ main() {
...
 	echo "done latest type"
+	
+	desc 'check if we restore data from binlogs'
+	kubectl patch pxc pitr --type=merge -p '{"spec":{"backup":{"pitr":{"enabled":true}}}}'
+	sleep 20
+	run_backup "$cluster" "on-pitr-minio1"
+	write_more_data "$cluster"
+	dest=$(sed 's,/,\\/,g' <<<$(kubectl get pxc-backup on-pitr-minio1 -o jsonpath='{.status.destination}'))
+	sleep 80 # need to wait while collector catch new data
+	run_recovery_check_pitr "$cluster" "restore-on-pitr-minio1" "on-pitr-minio1" "select-5" "" "$dest" ""
```
Comment (delgod) [inline e2e-tests/pitr/run:128]: why we need similar check as in 109 line? can we add `run_backup ` and `write_more_data` before 108 line?

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | passed |
| recreate | passed |
| storage | passed |
| haproxy | passed |
| big-data | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | passed |
| smart-update | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | failed |
| users | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/7f3c20756115b3e35773a1ee478f75437908414d
image: `perconalab/percona-xtradb-cluster-operator:PR-865-7f3c2075`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| scheduled-backup | passed |
| big-data | passed |
| recreate | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| limits | passed |
| self-healing-advanced | passed |
| upgrade-proxysql | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| affinity | passed |
| smart-update | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| pitr | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| upgrade-consistency | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/cdb35e2204a12b9e293fe4d11325ab686548292b
image: `perconalab/percona-xtradb-cluster-operator:PR-865-cdb35e22`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| haproxy | passed |
| upgrade-haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| storage | passed |
| affinity | passed |
| self-healing | passed |
| one-pod | passed |
| self-healing-advanced | passed |
| auto-tuning | passed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| big-data | passed |
| validation-hook | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | failed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/cdb35e2204a12b9e293fe4d11325ab686548292b
image: `perconalab/percona-xtradb-cluster-operator:PR-865-cdb35e22`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| recreate | passed |
| proxysql-sidecar-res-limits | passed |
| restore-to-encrypted-cluster | passed |
| users | passed |
| demand-backup | passed |
| tls-issue-self | passed |
| pitr | failed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| scaling-proxysql | passed |
| scheduled-backup | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/cdb35e2204a12b9e293fe4d11325ab686548292b
image: `perconalab/percona-xtradb-cluster-operator:PR-865-cdb35e22`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| big-data | passed |
| storage | passed |
| haproxy | passed |
| scheduled-backup | passed |
| recreate | passed |
| upgrade-haproxy | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| self-healing | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | passed |
| users | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/865/commits/aec14947cfa124836b7abbcc0d47c0c8848f1b66
image: `perconalab/percona-xtradb-cluster-operator:PR-865-aec14947`

**Review by tplavcic — DISMISSED** [review]

**Pull request merged**